src/annotator/highlighter/overlay-highlighter/index.js

```diff
@@ -1,18 +1,67 @@
 'use strict';
 
+const debounce = require('lodash.debounce');
+const OverlayHighlight = require('./overlay-highlight');
+const overlayEvents = require('./overlay-events');
+
+const DEBOUNCE_WAIT = 50;
+const _highlights = [];
```

Comment:
There is quite a lot of private state in this module, which I think is likely to make life difficult when writing tests because you haven't got an easy way to get a fresh state for each test and dispose of event handlers afterwards. It might be easier to wrap it up in a class.